WordPress/src/main/java/org/wordpress/android/ui/stats/StatsActivity.java

```diff
@@ -203,6 +206,12 @@ public void onItemSelected(AdapterView<?> parent, View view, int position, long
                     AppPrefs.setStatsTimeframe(mCurrentTimeframe);
                     mRequestedDate = StatsUtils.getCurrentDateTZ(mLocalBlogID);
                     createFragments(true); // Need to recreate fragment here, since a new timeline was selected.
+                    mSpinner.postDelayed(new Runnable() {
+                        @Override
+                        public void run() {
+                            scrollToTop();
```

Comment:
Don't think so, you need to rotate the device and change "timeperiod" at the same time.